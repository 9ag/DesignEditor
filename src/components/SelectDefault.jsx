import React from 'react'
const mobxReact = require('mobx-react')
import Select, { Option } from 'rc-select'

class SelectDefault extends React.Component {
  constructor(props) {
    super(props)
  }

  onChange = (e) => {
    let value
    if (e && e.target) {
      value = e.target.value
    } else {
      value = e
    }
    if (value) {
      const clickedItem = this.props.data.find(data => data.Id === value)
      this.props.choose(this.props.data, clickedItem)
    } else {
      this.props.choose(this.props.data, {})
    }
  }

  render() {
    const {data, placeholder} = this.props
    const selected = data.find(item => item._selected)
    const sValue = selected ? selected.Id : placeholder
    return (
      <div className="search-group">
        <Select
          value={sValue}
          placeholder="placeholder"
          dropdownMenuStyle={{ maxHeight: 200 }}
          style={{ width: '100%' }}
          allowClear
          optionLabelProp="children"
          optionFilterProp="text"
          onChange={this.onChange}
          firstActiveValue="2"
          backfill>
          {
            data.map(item => {
              return <Option value={item.Id} key={item.Id} text={item.Name}>
                {item.Name}
              </Option>
            })
          }
        </Select>
      </div>
    )
  }
}

export default mobxReact.observer(SelectDefault)