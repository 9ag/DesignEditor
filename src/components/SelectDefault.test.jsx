
import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

import SelectDefault from './SelectDefault';
describe("SelectDefault component test", () => {
    let choose = null;
    const data = [{
      Id: 1,
      Name: 'changsha',
      _selected: true,
    }, {
      Id: 2,
      Name: 'xiangtan',
    }, {
      Id: 3,
      Name: 'loudi',
    }];
    beforeEach(() => {
      choose = jest.fn();
    });

    it("choose callback test", () => {
      const props = {
        placeholder: "Filter by Responsible Vendor",
        data,
        choose,
      };
      const component = shallow(<SelectDefault {...props} />);
      component.find('Select').simulate('change', {
        preventDefault() {},
        target: { value: 3 }
      });
      expect(choose).toBeCalledWith(data, data[2]);
      component.find('Select').simulate('change', 3);
      expect(choose).toBeCalledWith(data, data[2]);
      component.find('Select').simulate('change', null);
      expect(choose).toBeCalledWith(data, {});
    });
     
    it('snapshot test', () => {
      const props = {
        placeholder: "Filter by Responsible Vendor",
        data,
        choose,
      };
      const component = renderer.create(<SelectDefault {...props} />).toJSON();
      expect(component).toMatchSnapshot();
    });
});