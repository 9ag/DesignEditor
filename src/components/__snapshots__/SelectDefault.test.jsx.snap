// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDefault component test snapshot test 1`] = `
<div
  className="search-group"
>
  <div
    className="rc-select rc-select-enabled rc-select-allow-clear"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup="true"
      className="rc-select-selection
            rc-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="rc-select-selection__rendered"
      >
        <div
          className="rc-select-selection__placeholder"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "display": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          placeholder
        </div>
        <div
          className="rc-select-selection-selected-value"
          style={
            Object {
              "display": "block",
              "opacity": 1,
            }
          }
          title="changsha"
        >
          changsha
        </div>
        <div
          className="rc-select-search rc-select-search--inline"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="rc-select-search__field__wrap"
          >
            <input
              autoComplete="off"
              className="rc-select-search__field"
              onChange={[Function]}
              onKeyDown={[Function]}
              value=""
            />
            <span
              className="rc-select-search__field__mirror"
            >
              
              Â 
            </span>
          </div>
        </div>
      </div>
      <span
        className="rc-select-selection__clear"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      />
      <span
        className="rc-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <b />
      </span>
    </div>
  </div>
</div>
`;
